p2p/src/net/default_backend/mod.rs

```diff
@@ -60,6 +60,21 @@ pub struct ConnectivityHandle<S: NetworkingService, T: TransportSocket> {
     _marker: PhantomData<fn() -> S>,
```

Comment:
I just noticed this... why not `PhantomData<S>`? It has the same variance.

Reply:
I thought about changing it, but decided that `fn()` is there for a reason :smile: 
I also removed it from a few other places too.

Reply:
Maybe there's a reason I'm not remembering... I vaguely am now recalling there's a reason other than variance... but can't exactly remember.

Reply:
@iljakuklic Do you remember why `PhantomData<fn() -> S>` was used instead `PhantomData<S>`?
The code seems to build normally with the later.

Reply:
I reverted my changes.